indykite/resource_entity_matching_pipeline.go

```diff
@@ -82,16 +98,25 @@ func resourceEntityMatchingPipelineFlatten(
 	var d diag.Diagnostics
 	entitymatching := resp.GetConfigNode().GetEntityMatchingPipelineConfig()
 
-	sourceTypes := make([]string, len(entitymatching.NodeFilter.GetSourceNodeTypes()))
-	for i, source := range entitymatching.NodeFilter.GetSourceNodeTypes() {
-		sourceTypes[i] = source
+	if len(entitymatching.NodeFilter.GetSourceNodeTypes()) > 0 {
+		sourceTypes := make([]string, len(entitymatching.NodeFilter.GetSourceNodeTypes()))
+		for i, source := range entitymatching.NodeFilter.GetSourceNodeTypes() {
+			sourceTypes[i] = source
+		}
+		setData(&d, data, entityMatchingPipelineSourceNodeFilterKey, sourceTypes)
 	}
-	setData(&d, data, entityMatchingPipelineSourceNodeFilterKey, sourceTypes)
-	targetTypes := make([]string, len(entitymatching.NodeFilter.GetTargetNodeTypes()))
-	for i, target := range entitymatching.NodeFilter.GetTargetNodeTypes() {
-		targetTypes[i] = target
+	if len(entitymatching.NodeFilter.GetTargetNodeTypes()) > 0 {
```

Comment:
Same here, `if` can be removed I believe.

Reply:
yes indeed because when we update in fact  we delete and create